src/snakemake/unit_tests/templates/ruletest.py.jinja2

```diff
@@ -1,56 +1,71 @@
+import subprocess as sp
+from pathlib import Path
 
 sys.path.insert(0, os.path.dirname(__file__))
 
-import common
-
 
-def test_{{ ruletest.name }}():
+def test_{{ ruletest.name }}(conda_prefix):
 
-    with TemporaryDirectory() as tmpdir:
+    with tempfile.TemporaryDirectory() as tmpdir:
         workdir = Path(tmpdir) / "workdir"
-        data_path = PurePosixPath("{{ ruletest.data_path.as_posix() }}")
-        expected_path = PurePosixPath("{{ ruletest.expected_path.as_posix() }}")
+        data_path = Path("{{ ruletest.data_path.as_posix() }}")
+        expected_path = Path("{{ ruletest.expected_path.as_posix() }}")
 
         # Copy data to the temporary workdir.
         shutil.copytree(data_path, workdir)
 
-        # dbg
-        print("{{ ruletest.target }}", file=sys.stderr)
+        # Copy config/ (if exists) to the temporary workdir.
+        config = Path("config")
+        if config.exists() and config.is_dir():
+            shutil.copytree("config", workdir / config, dirs_exist_ok=True)
 
         # Run the test job.
-        sp.check_output([
-            "python",
-            "-m",
-            "snakemake", 
-            "{{ ruletest.target }}",
-            "-f", 
-            "-j1",
-            "--target-files-omit-workdir-adjustment",
-            {% if configfiles %}
-            "--configfile",
-            {% for configfile in configfiles %}
-            {{ configfile }}
-            {% endfor %}
-            {% endif %}    
-            {% if "conda" in deploy %}
-            "--use-conda",
-            {% endif %}
-            {% if "singularity" in deploy %}
-            "--use-singularity",
-            {% endif %}
-            "--directory",
-            workdir,
-        ])
-
-        # Check the output byte by byte using cmp.
+        sp.check_output(
+            [
+                "python",
+                "-m",
+                "snakemake",
+                {% for target in ruletest.target %}
+                "{{ target }}",
+                {% endfor %}
+                {% if snakefile %}
+                "--snakefile",
+                "{{ snakefile }}",
+                {% endif %}
+                "-f",
+                "--notemp",
+                "-j1",
+                "--target-files-omit-workdir-adjustment",
+                {% if configfiles %}
+                "--configfile",
+                {% for configfile in configfiles %}
+                "{{ configfile }}",
+                {% endfor %}
+                {% endif %}
+                {% if deploy %}
+                "--software-deployment-method",
+                {% for sdm in deploy %}
+                "{{ sdm }}",
+                {% endfor %}
+                {% endif %}
+                "--directory",
+                workdir,
+            ]
+            + conda_prefix
+        )
+
+        # Check the output byte by byte using cmp/zmp/bzcmp.
         # To modify this behavior, you can inherit from common.OutputChecker in here
```

Comment:
Perhaps adding a pytest option like the conda prefix to modify this behavior so that one does not have to edit each test file?

Reply:
Maybe I should explain it better in the docs, but the idea is that the user can provide `--conda-prefix` through the command line, e.g.:
```
pytest .tests/unit --conda-prefix /path/to/conda/envs
```